tools/python/boutiques/dataHandler.py

```diff
@@ -0,0 +1,276 @@
+        self.cache_files = os.listdir(self.cache_dir)
+        self.descriptor_files = [fl for fl in self.cache_files
+                                 if fl.split('-')[0] == "descriptor"]
+        self.record_files = [fl for fl in self.cache_files
+                             if fl not in self.descriptor_files]
+
+    # Function to display the contents of the cache
+    # Option to display an example file which displays an pre-made example file
+    # or the first file of the cache if it exists
+    # Otherwise displays information about the cache and a list of files
+    def inspect(self, example=False):
+        self.example = example
+        # Display an example json file
+        if self.example:
+            # Display the first file in cache
+            if len(self.cache_files) > 0:
+                filename = self.cache_files[0]
+                file_path = os.path.join(self.cache_dir, filename)
+                self._display_file(file_path)
+            # Display the example file
+            else:
+                example_fn = os.path.join(bfile, "examples", "execution-data",
+                                          "execution-record.json")
+                self._display_file(example_fn)
+        # Print information about files in cache
+        else:
+            print("There are {} unpublished records in the cache"
+                  .format(len(self.record_files)))
+            print("There are {} unpublished descriptors in the cache"
+                  .format(len(self.descriptor_files)))
+            for i in range(len(self.cache_files)):
+                print(self.cache_files[i])
+
+    # Private function to print a file to console
+    def _display_file(self, file_path):
+        with open(file_path, 'r') as file_in:
+            print(file_in.read())
+            file_in.close()
+
+    # Function to publish a data set to Zenodo
+    # Options allow to only publish a single file, publish files individually as
+    # data sets or bulk publish all files in the cache as one data set (default)
+    def publish(self, file, zenodo_token, author,
+                individually=False, sandbox=False, no_int=False,
+                verbose=False):
+        self.filename = extractFileName(file)
+        self.zenodo_access_token = zenodo_token
+        self.author = "Anonymous"
+        if author is not None:
+            self.author = author
+        self.individual = individually
+        self.sandbox = sandbox
+        self.no_int = no_int
+        self.verbose = verbose
+        self.zenodo_helper = ZenodoHelper(sandbox, no_int, verbose)
+        self.zenodo_endpoint = self.zenodo_helper.get_zenodo_endpoint()
+
+        # Set Zenodo access token and save to configuration file
+        if self.zenodo_access_token is None:
+            self.zenodo_access_token = self.zenodo_helper\
+                .get_zenodo_access_token()
+        self.zenodo_helper.save_zenodo_access_token(self.zenodo_access_token)
```

Comment:
This is also in the publisher, it should be factorized in ZenodoHelper.

Reply:
refactored